project/ValidatePullRequest.scala

```diff
@@ -30,22 +30,21 @@ object ValidatePullRequest extends AutoPlugin {
       l.info(s"Skipping validation of [$projectName], as PR does NOT affect this project...")
   }
   case object BuildQuick extends BuildMode {
-    override def task = Zero.dependsOn(test in ValidatePR)
-    def log(projectName: String, l: Logger) = 
+    override def task = Zero.dependsOn(test in ValidatePR).dependsOn(unidoc in Compile)
+    def log(projectName: String, l: Logger) =
       l.info(s"Building [$projectName] in quick mode, as it's dependencies were affected by PR.")
   }
   case object BuildProjectChangedQuick extends BuildMode {
-    override def task = Zero.dependsOn(test in ValidatePR)
+    override def task = Zero.dependsOn(test in ValidatePR).dependsOn(unidoc in Compile)
     def log(projectName: String, l: Logger) =
       l.info(s"Building [$projectName] as the root `project/` directory was affected by this PR.")
   }
   final case class BuildCommentForcedAll(phrase: String, c: GHIssueComment) extends BuildMode {
-    override def task = Zero.dependsOn(test in Test)
+    override def task = Zero.dependsOn(test in Test).dependsOn(unidoc in Compile)
```

Comment:
For this to work add `scalaJavaUnidocSettings` to `projectSettings` on line 122, otherwise it's a reference to undefined setting during the run. 
After fixing this I got the "duplicated class" problems from genjavadoc (I assume, looks like it) – we're almost there it seems.